lib/hal_client.rb

```diff
@@ -175,7 +176,7 @@ def interpret_response(resp)
       begin
         Representation.new(hal_client: self, parsed_json: MultiJson.load(resp.to_s),
                            href: location)
-      rescue MultiJson::ParseError, InvalidRepresentationError => e
+      rescue MultiJson::ParseError, InvalidRepresentationError, SyntaxError => e
```

Comment:
This isn't really caused by malformed ruby code -- it's a poorly chosen error that MultiJson is throwing:
https://github.com/intridea/multi_json/blob/1ba3be42600696f676d9cc1f029227b84c3c763b/lib/multi_json/adapters/oj.rb#L11

It looks like MultiJson assigns the constant 'SyntaxError' to the constant MultiJson::Adapters::Oj::ParseError as a fallback if ::Oj::ParseError is not defined. We don't seem to be able to catch this exception if we try to rescue MultiJson::Adapters::Oj::ParseError, but we do catch it if we rescue SyntaxError.

Not sure how best to handle this. More suggestions/insights welcome :)